### PR TITLE
Feat/add model performance logging

### PR DESCRIPTION
**Pull request by austinagii**
Implements wandb logging during training.

The following metrics are logged:

Training:
- Cross Entropy Loss
- Perplexity 
- Gradient norm
- Learning Rate

Validation:
- Cross Entropy Loss
- Perplexity

The following are also included:
- Allow wandb logging to be disabled from the command line
- Resume wandb runs when training is resumed from a checkpoint

### CODE REVIEW

File: scripts/train.py [hunk 1]
```diff
@@ -251,13 +242,41 @@ def main(
...
+
     # Start the training loop.
     for epoch in itertools.count(current_epoch + 1):
         train_state = train(
+            run=wandb_run,
             model=model,
             tokenizer=tokenizer,
             dataset=train_dataset,
             current_epoch=epoch,
+            global_step=global_step,
```
[the remark this answers is not in the document]
Comment (austinagii) [reply scripts/train.py:279]: Fixed
